examples/cryptocurrency-advanced/frontend/src/plugins/blockchain.js

```diff
@@ -4,200 +4,142 @@ import * as proto from '../../proto/stubs.js'
+  for (const tx of txTypes) {
+    const txData = tx.deserialize(Exonum.hexadecimalToUint8Array(transaction))
+    if (txData) return Object.assign({}, txData.payload, {
+      hash: txData.hash(),
+      to: txData.payload.to ? Exonum.uint8ArrayToHexadecimal(txData.payload.to.data) : undefined
+    })
   }
-
-  if (transaction.to) {
-    return new TransferTransaction(publicKey)
-  }
-
-  return new IssueTransaction(publicKey)
+  return { name: 'initialTx' }
 }
 
+
 module.exports = {
-  install(Vue) {
+  install (Vue) {
     Vue.prototype.$blockchain = {
-      generateKeyPair() {
+      generateKeyPair () {
         return Exonum.keyPair()
       },
 
-      generateSeed() {
+      generateSeed () {
         return Exonum.randomUint64()
       },
 
-      createWallet(keyPair, name) {
-        // Describe transaction
-        const transaction = new CreateTransaction(keyPair.publicKey)
-
-        // Transaction data
-        const data = {
-          name: name
-        }
-
+      createWallet (keyPair, name) {
+        const walletTx = WalletTx()
+        const transaction = walletTx.create({ name }, keyPair).serialize()
         // Send transaction into blockchain
-        return transaction.send(TRANSACTION_URL, data, keyPair.secretKey)
+        return Exonum.send(TRANSACTION_URL, transaction)
       },
 
-      addFunds(keyPair, amountToAdd, seed) {
+      addFunds (keyPair, amountToAdd, seed) {
         // Describe transaction
-        const transaction = new IssueTransaction(keyPair.publicKey)
+        const issueTx = IssueTransaction()
 
         // Transaction data
         const data = {
           amount: amountToAdd.toString(),
           seed: seed
         }
+        const transaction = issueTx.create(data, keyPair).serialize()
 
         // Send transaction into blockchain
-        return transaction.send(TRANSACTION_URL, data, keyPair.secretKey)
+        return Exonum.send(TRANSACTION_URL, transaction)
       },
 
-      transfer(keyPair, receiver, amountToTransfer, seed) {
+      transfer (keyPair, receiver, amountToTransfer, seed) {
         // Describe transaction
-        const transaction = new TransferTransaction(keyPair.publicKey)
+        const transferTx = TransferTransaction()
 
         // Transaction data
         const data = {
-          to: { data: Exonum.hexadecimalToUint8Array(receiver) },
+          to: { data: Exonum.hexadecimalToUint8Array(Exonum.publicKeyToAddress(receiver)) },
           amount: amountToTransfer,
           seed: seed
         }
+        console.log(data)
+        const transaction = transferTx.create(data, keyPair).serialize()
 
         // Send transaction into blockchain
-        return transaction.send(TRANSACTION_URL, data, keyPair.secretKey)
+        return Exonum.send(TRANSACTION_URL, transaction)
       },
 
-      getWallet(publicKey) {
-        return axios.get('/api/services/configuration/v1/configs/actual').then(response => {
+      getWallet (publicKey) {
```

Comment:
Is `async` / `await` inapplicable here? Could significantly remove readability, IMO.

Reply:
The whole application on promises, for now, this why I stick this way. But in the future, this should be on async/await for the whole project - for sure.